src/hatch_fancy_pypi_readme/_config.py

```diff
@@ -19,36 +22,97 @@ class Config:
     substitutions: list[Substituter]
 
 
+SCHEMA = {
+    "type": "object",
+    "properties": {
+        "content-type": {
+            "type": "string",
+            "enum": ["text/markdown", "text/x-rst"],
+        },
+        "fragments": {
+            "type": "array",
+            "minItems": 1,
+            # Items are validated separately for better error messages.
+            "items": {"type": "object"},
+        },
+        "substitutions": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "pattern": {"type": "string", "format": "regex"},
+                    "replacement": {"type": "string"},
+                    "ignore-case": {"type": "boolean"},
+                },
+                "required": ["pattern", "replacement"],
+                "additionalProperties": False,
+            },
+        },
+    },
+    "required": ["content-type", "fragments"],
+    "additionalProperties": False,
+}
+
+V = Draft202012Validator(
+    SCHEMA, format_checker=Draft202012Validator.FORMAT_CHECKER
+)
+
+
 def load_and_validate_config(config: dict[str, Any]) -> Config:
-    errs = []
+    errs = sorted(
```

Comment:
If you want all errors, and to have them in some useful order, usually the first thing to try is `sorted(V.iter_errors(config), key=jsonschema.exceptions.relevance)` -- but sometimes that's not smart enough yet (e.g. https://github.com/python-jsonschema/jsonschema/issues/698 is still open which may happen here). But worth a shot if you didn't try it first.

Reply:
this should maybe be added to the top of the validation docs I got `key=lambda e: e.path` from there

Reply:
That's needed there strictly because otherwise doctests would be flaky :/ (and indeed `relevance` isn't desirable there since its order would possibly change). But I can look at how to hide that better from readers.

Reply:
or just add a "you probably want relevance, check #here"

it's fine to add helpful cross-references to docs :)